Base/QTGUI/qSlicerSingletonViewFactory.h

```diff
@@ -0,0 +1,68 @@
+/*==============================================================================
+
+Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+See COPYRIGHT.txt
+or http://www.slicer.org/copyright/copyright.txt for details.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+and was supported through CANARIE's Research Software Program, and Cancer
+Care Ontario.
+
+==============================================================================*/
+
+#ifndef qSlicerSingletonViewFactory_h
+#define qSlicerSingletonViewFactory_h
+
+// CTK includes
+#include "ctkLayoutViewFactory.h"
+
+// QtGUI includes
+#include "qSlicerBaseQTGUIExport.h"
+
+class ctkDICOMBrowser;
+class qSlicerSingletonViewFactoryPrivate;
+
+/// This class provides an interface for C++ and Python classes to register a singleton view replacement widget.
+/// New view widgets can be registered by registering a new qSlicerSingletonViewFactory, and then setting the widget and tag using setWidget(QWidget*) and
+/// setTagName(QString).
+/// This factory contains a single pointer to an instance of the widget, so only one view can be created within a given layout.
```

Comment:
What happens if two instances are used in a layout? Could we check this and report an error if it happens?

Reply:
If two or more of the same singleton view are included in the same layout, then only the last one will be filled. Previous views will be blank.

Probably could log an error if this happens.

Reply:
If two of the same views are created within the same layout, the following error message is now displayed:
````
qSlicerSingletonViewFactory::createViewFromXML - Widget for view "TagName" is already in use within the current layout!
````